src/setuptools_scm/version.py

```diff
@@ -309,6 +309,69 @@ def no_guess_dev_version(version):
         return version.format_with("{tag}.post1.dev{distance}")
 
 
+def date_ver_match(ver):
+    match = re.match(
+        (
+            r"^(?P<date>(?P<year>\d{2}|\d{4})(?:\.\d{1,2}){2})"
+            r"(?:\.(?P<patch>\d*)){0,1}?$"
+        ),
+        str(ver),
+    )
+    return match
+
+
+def guess_next_date_ver(version, node_date=None):
```

Comment:
I think it doesn't because the `format_next_version` function only passes the version tag, rather than the full version object. Maybe I should be using `format_next_version` to call `guess_next_date_ver`?

Reply:
ah yes, please use that, format_next_version was made exactly for that use-case